src/editor/Renderer.cpp

```diff
@@ -1529,6 +1593,41 @@ void Renderer::getPickRay(vec3& start, vec3& pickDir) {
 	pickDir = (start - cameraOrigin).normalize(1.0f);
 }
 
+BspRenderer* Renderer::getSelectedMap() {
+	for (int i = 0; i < mapRenderers.size(); i++)
+	{
+		BspRenderer* s = mapRenderers[i];
+		if (s->map && s->map->is_selected)
+		{
+			return s;
+		}
+	}
+	return mapRenderers.size() ? mapRenderers[0] : NULL;
+}
+
+int Renderer::getSelectedMapId() {
+	for (int i = 0; i < mapRenderers.size(); i++)
+	{
+		BspRenderer* s = mapRenderers[i];
+		if (s->map && s->map->is_selected)
+		{
+			return i;
+		}
+	}
+	return 0;
+}
+
+void Renderer::setSelectedMap(int id) {
+	for (int i = 0; i < mapRenderers.size(); i++)
+	{
+		BspRenderer* s = mapRenderers[i];
+		if (s->map)
+		{
+			mapRenderers[id]->map->is_selected = i == id;
+		}
+	}
+}
+

```

Comment:
the `pickInfo` field can be used instead, once the `mapIdx` change is reverted.

Reply:
I just remove mapidx for easy access maps, and preparing for multiple map edit